pkg/client/client.go

```diff
@@ -61,14 +64,29 @@ func NewClient(address string, certFile string, hmacKey string, logger *zap.Logg
 	// Set up a connection to the server.
 	conn, err := grpc.Dial(address, opts...)
 	if err != nil {
-		logger.Sugar().Errorf("did not connect: %v", err)
 		return nil, err
 	}
 	client.conn = conn
 
 	return client, nil
 }
 
+// NewClientWithRetry tries creating a new client for the services for the given address, with the certificate and hmac until successfully initialized.
+func NewClientWithRetry(address string, certFile string, hmacKey string, logger *zap.Logger, stop <-chan struct{}) (Client, error) {
```

Comment:
Where should the stop channel come from ?

Reply:
It can be nil as well, it's just if the caller wants to leave the retry loop.